### PR TITLE
feat: support self-host

### PR DESCRIPTION
**Pull request by helsonxiao**
Closes issue: 

## 📜 Summary of changes:

This PR provides some support for self-host:

- `browser-extension/config/configs/env/custom.json` can be used to customize the domain name, this file will not be tracked by git
- Ignore `.firebaserc`, `firebase.json`, `*.cache files`, these files are needed when executing `firebase deploy` locally
- Added environment variable `REACT_APP_FIREBASE_OAUTH_CLIENT_ID` to switch OAuth service accounts

Known limitations:
- Unable to use proprietary team features
- Firebase’s Spark plan does not provide cloud function services, so you can’t try out features that depend on cloud functions for free
- The Requestly team may introduce code for team features into the main process during iterations, causing the personal version to not work properly, such as:
  - https://github.com/requestly/requestly/pull/1211
    Currently, my temporary solution is to comment out this line.
https://github.com/requestly/requestly/blob/cb6de43b4add6adbe5ef331c84e9dc902e40e54f/app/src/hooks/AuthHandler.js#L60

## ✅ Checklist:

- [ ] Make sure linting and unit tests pass.
- [ ] No install/build warnings introduced.
- [ ] Verified UI in browser.
- [ ] For UI changes, added/updated analytics events (if applicable).
- [ ] For changes in extension's code, manually tested in Chrome and Firefox.
- [ ] For changes in extension's code, both MV2 and MV3 are covered.
- [ ] Added/updated unit tests for this change.
- [ ] Raised pull request to update corresponding documentation (if already exists).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -16,7 +16,11 @@ app/src/config/pricing/plans/plans.json
 app/src/config/pricing/features/features.json
 app/.env.local
 !app/public/*
+browser-extension/config/configs/env/custom.json
 .DS_Store
```
Comment (demoManito) [inline .gitignore:20]: 这个重复了

**Comment by sagarsoni7** [comment]
Thanks @ helsonxiao & @demoManito for your contribution. Greatly appreciated

**Comment by wrongsahil** [comment]
@helsonxiao I don't think this is enough for self-hosting. Lots of other things needs to be changed. We'll pick this internally. Thanks for the contribution though 🙌

**Comment by helsonxiao** [comment]
@wrongsahil As I have described, there are indeed still some limitations that require your unified handling. May I ask if there is a development plan in place?

**Comment by wrongsahil** [comment]
In self hosted version, proprietary features won't be available. But rest of the things should work without any limitations. We'll try to make the process as smooth as possible.

While this isn't our immediate priority, but we'll definitely be supporting this in future. I'll open a issue for this

**Comment by helsonxiao** [comment]
Sure, that's great!

**Pull request closed without merging**